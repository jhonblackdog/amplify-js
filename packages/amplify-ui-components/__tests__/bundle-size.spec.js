// @ts-nocheck
/* eslint-disable @typescript-eslint/no-var-requires */
const { spawnSync } = require('child_process');
const globby = require('globby');
const path = require('path');

const dist = path.resolve(__dirname, '../dist');

// See: https://stackoverflow.com/a/18650828
function formatBytes(bytes, decimals = 0) {
  if (bytes === 0) return '0 Bytes';

  const k = 1024;
  const dm = decimals < 0 ? 0 : decimals;
  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
  const i = Math.floor(Math.log(bytes) / Math.log(k));

  return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
}

describe('@amplify/ui-components bundle size', () => {
  beforeAll(() => {
    spawnSync('yarn', ['build'], {
      cwd: path.resolve(__dirname, '..'),
      stdio: 'inherit',
    });
  });

  it('should match snapshot', async () => {
    const stats = await globby('**', { cwd: dist, stats: true });
    const sizes = stats.reduce((acc, stat) => {
      acc[stat.path] = formatBytes(stat.size);
      return acc;
    }, {});

    expect(sizes).toMatchSnapshot();
  });
});
