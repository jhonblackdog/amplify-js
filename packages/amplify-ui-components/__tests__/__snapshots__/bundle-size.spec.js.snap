// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@amplify/ui-components bundle size should match snapshot 1`] = `
Object {
  "amplify-ui-components.js": "1 KB",
  "amplify-ui-components/amplify-ui-components.esm.js": "6 KB",
  "amplify-ui-components/amplify-ui-components.js": "121 KB",
  "amplify-ui-components/index.esm.js": "91 Bytes",
  "amplify-ui-components/p-006494f7.js": "3 KB",
  "amplify-ui-components/p-08b6a7d5.system.js": "739 Bytes",
  "amplify-ui-components/p-0a3ab270.js": "601 Bytes",
  "amplify-ui-components/p-0e6cd8db.js": "8 KB",
  "amplify-ui-components/p-2c5b4821.js": "13 KB",
  "amplify-ui-components/p-3baa856a.js": "1 KB",
  "amplify-ui-components/p-3ccc407f.system.js": "6 KB",
  "amplify-ui-components/p-446995ed.js": "18 KB",
  "amplify-ui-components/p-5e0662bc.system.js": "6 KB",
  "amplify-ui-components/p-6f736fb7.js": "1 KB",
  "amplify-ui-components/p-7be3472e.system.js": "19 KB",
  "amplify-ui-components/p-7d872662.system.js": "3 KB",
  "amplify-ui-components/p-8f474250.js": "560 Bytes",
  "amplify-ui-components/p-92ec8516.system.js": "8 KB",
  "amplify-ui-components/p-9c77107f.system.js": "21 KB",
  "amplify-ui-components/p-9r2mtm0j.system.entry.js": "882 Bytes",
  "amplify-ui-components/p-a0bdb6be.system.js": "1 KB",
  "amplify-ui-components/p-acab45df.system.js": "203 Bytes",
  "amplify-ui-components/p-bcmqnvi8.entry.js": "673 Bytes",
  "amplify-ui-components/p-bukjagyf.entry.js": "113 KB",
  "amplify-ui-components/p-ecad7cb1.system.js": "1 KB",
  "amplify-ui-components/p-eccjkkrs.system.entry.js": "3 KB",
  "amplify-ui-components/p-f552c273.system.js": "733 Bytes",
  "amplify-ui-components/p-f792bbf6.js": "5 KB",
  "amplify-ui-components/p-jrjb60x5.entry.js": "1 KB",
  "amplify-ui-components/p-lozc2zyv.entry.js": "998 Bytes",
  "amplify-ui-components/p-lp5d5fht.system.entry.js": "132 KB",
  "amplify-ui-components/p-rbmlv527.system.entry.js": "1 KB",
  "amplify-ui-components/p-sqnajrh2.system.entry.js": "2 KB",
  "amplify-ui-components/p-zye1olmm.entry.js": "1006 Bytes",
  "cjs/Translations-5faaa099.js": "4 KB",
  "cjs/amplify-amazon-button_44.cjs.entry.js": "232 KB",
  "cjs/amplify-checkbox.cjs.entry.js": "1 KB",
  "cjs/amplify-federated-sign-in.cjs.entry.js": "2 KB",
  "cjs/amplify-greetings.cjs.entry.js": "2 KB",
  "cjs/amplify-nav_2.cjs.entry.js": "2 KB",
  "cjs/amplify-ui-components.cjs.js": "6 KB",
  "cjs/app-globals-a1567b1f.js": "2 KB",
  "cjs/auth-types-cbf3c8e1.js": "2 KB",
  "cjs/constants-c59bf284.js": "2 KB",
  "cjs/css-shim-d0251953.js": "10 KB",
  "cjs/dom-df31b462.js": "19 KB",
  "cjs/helpers-0971d472.js": "1 KB",
  "cjs/index-eaa6275b.js": "70 KB",
  "cjs/index.cjs.js": "450 Bytes",
  "cjs/loader.cjs.js": "7 KB",
  "cjs/shadow-css-771cb161.js": "15 KB",
  "collection/collection-manifest.json": "3 KB",
  "collection/common/Translations.js": "4 KB",
  "collection/common/constants.js": "1 KB",
  "collection/common/country-dial-codes.js": "8 KB",
  "collection/common/helpers.js": "1 KB",
  "collection/common/testing.js": "2 KB",
  "collection/common/types/auth-types.js": "2 KB",
  "collection/common/types/ui-types.js": "0 Bytes",
  "collection/components/amplify-amazon-button/amplify-amazon-button.js": "5 KB",
  "collection/components/amplify-auth-fields/amplify-auth-fields-interface.js": "0 Bytes",
  "collection/components/amplify-auth-fields/amplify-auth-fields.js": "2 KB",
  "collection/components/amplify-auth-fields/component-field-mapping.js": "2 KB",
  "collection/components/amplify-auth0-button/amplify-auth0-button.js": "5 KB",
  "collection/components/amplify-authenticator/amplify-authenticator.js": "9 KB",
  "collection/components/amplify-button/amplify-button.css": "1 KB",
  "collection/components/amplify-button/amplify-button.js": "4 KB",
  "collection/components/amplify-checkbox/amplify-checkbox.css": "256 Bytes",
  "collection/components/amplify-checkbox/amplify-checkbox.js": "4 KB",
  "collection/components/amplify-code-field/amplify-code-field.js": "6 KB",
  "collection/components/amplify-confirm-sign-in/amplify-confirm-sign-in.js": "10 KB",
  "collection/components/amplify-confirm-sign-up/amplify-confirm-sign-up.js": "10 KB",
  "collection/components/amplify-country-dial-code/amplify-country-dial-code-interface.js": "0 Bytes",
  "collection/components/amplify-country-dial-code/amplify-country-dial-code.js": "3 KB",
  "collection/components/amplify-email-field/amplify-email-field.js": "6 KB",
  "collection/components/amplify-examples/amplify-examples.js": "1 KB",
  "collection/components/amplify-examples/auth-totp-example.js": "584 Bytes",
  "collection/components/amplify-examples/authenticator-example.js": "229 Bytes",
  "collection/components/amplify-examples/custom-authenticator-example.js": "435 Bytes",
  "collection/components/amplify-examples/example4.js": "453 Bytes",
  "collection/components/amplify-examples/rock-paper-scissor/rock-paper-scissor.css": "69 Bytes",
  "collection/components/amplify-examples/rock-paper-scissor/rock-paper-scissor.js": "2 KB",
  "collection/components/amplify-examples/scene-example.js": "191 Bytes",
  "collection/components/amplify-facebook-button/amplify-facebook-button.js": "5 KB",
  "collection/components/amplify-federated-buttons/amplify-federated-buttons.js": "5 KB",
  "collection/components/amplify-federated-sign-in/amplify-federated-sign-in.js": "4 KB",
  "collection/components/amplify-forgot-password/amplify-forgot-password-interface.js": "0 Bytes",
  "collection/components/amplify-forgot-password/amplify-forgot-password.js": "12 KB",
  "collection/components/amplify-form-field/amplify-form-field.css": "342 Bytes",
  "collection/components/amplify-form-field/amplify-form-field.js": "9 KB",
  "collection/components/amplify-form-section/amplify-form-section.css": "608 Bytes",
  "collection/components/amplify-form-section/amplify-form-section.js": "6 KB",
  "collection/components/amplify-google-button/amplify-google-button.js": "5 KB",
  "collection/components/amplify-greetings/amplify-greetings.css": "348 Bytes",
  "collection/components/amplify-greetings/amplify-greetings.js": "4 KB",
  "collection/components/amplify-hint/amplify-hint.css": "272 Bytes",
  "collection/components/amplify-hint/amplify-hint.js": "445 Bytes",
  "collection/components/amplify-icon-button/amplify-icon-button.css": "932 Bytes",
  "collection/components/amplify-icon-button/amplify-icon-button.js": "3 KB",
  "collection/components/amplify-icon/amplify-icon.css": "95 Bytes",
  "collection/components/amplify-icon/amplify-icon.js": "2 KB",
  "collection/components/amplify-icon/icons.js": "14 KB",
  "collection/components/amplify-input/amplify-input.css": "626 Bytes",
  "collection/components/amplify-input/amplify-input.js": "6 KB",
  "collection/components/amplify-label/amplify-label.css": "159 Bytes",
  "collection/components/amplify-label/amplify-label.js": "1006 Bytes",
  "collection/components/amplify-link/amplify-link.css": "423 Bytes",
  "collection/components/amplify-link/amplify-link.js": "1 KB",
  "collection/components/amplify-loading-spinner/amplify-loading-spinner.css": "227 Bytes",
  "collection/components/amplify-loading-spinner/amplify-loading-spinner.js": "493 Bytes",
  "collection/components/amplify-nav/amplify-nav.css": "125 Bytes",
  "collection/components/amplify-nav/amplify-nav.js": "440 Bytes",
  "collection/components/amplify-oauth-button/amplify-oauth-button.js": "1 KB",
  "collection/components/amplify-password-field/amplify-password-field.js": "6 KB",
  "collection/components/amplify-phone-field/amplify-phone-field.css": "157 Bytes",
  "collection/components/amplify-phone-field/amplify-phone-field.js": "7 KB",
  "collection/components/amplify-radio-button/amplify-radio-button.css": "241 Bytes",
  "collection/components/amplify-radio-button/amplify-radio-button.js": "5 KB",
  "collection/components/amplify-require-new-password/amplify-require-new-password.js": "9 KB",
  "collection/components/amplify-scene-loading/amplify-scene-loading.css": "1 KB",
  "collection/components/amplify-scene-loading/amplify-scene-loading.js": "9 KB",
  "collection/components/amplify-scene/amplify-scene-interface.js": "0 Bytes",
  "collection/components/amplify-scene/amplify-scene.css": "290 Bytes",
  "collection/components/amplify-scene/amplify-scene.js": "7 KB",
  "collection/components/amplify-section/amplify-section.css": "421 Bytes",
  "collection/components/amplify-section/amplify-section.js": "1 KB",
  "collection/components/amplify-select-mfa-type/amplify-select-mfa-type.js": "7 KB",
  "collection/components/amplify-select/amplify-select-interface.js": "0 Bytes",
  "collection/components/amplify-select/amplify-select.css": "1 KB",
  "collection/components/amplify-select/amplify-select.js": "3 KB",
  "collection/components/amplify-sign-in-button/amplify-sign-in-button.css": "2 KB",
  "collection/components/amplify-sign-in-button/amplify-sign-in-button.js": "1 KB",
  "collection/components/amplify-sign-in/amplify-sign-in-interface.js": "0 Bytes",
  "collection/components/amplify-sign-in/amplify-sign-in.css": "414 Bytes",
  "collection/components/amplify-sign-in/amplify-sign-in.js": "16 KB",
  "collection/components/amplify-sign-out/amplify-sign-out.js": "3 KB",
  "collection/components/amplify-sign-up/amplify-sign-up-interface.js": "0 Bytes",
  "collection/components/amplify-sign-up/amplify-sign-up.css": "401 Bytes",
  "collection/components/amplify-sign-up/amplify-sign-up.js": "17 KB",
  "collection/components/amplify-strike/amplify-strike.css": "436 Bytes",
  "collection/components/amplify-strike/amplify-strike.js": "464 Bytes",
  "collection/components/amplify-toast/amplify-toast.css": "2 KB",
  "collection/components/amplify-toast/amplify-toast.js": "2 KB",
  "collection/components/amplify-tooltip/amplify-tooltip.css": "2 KB",
  "collection/components/amplify-tooltip/amplify-tooltip.js": "2 KB",
  "collection/components/amplify-totp-setup/amplify-totp-setup-interface.js": "0 Bytes",
  "collection/components/amplify-totp-setup/amplify-totp-setup.css": "113 Bytes",
  "collection/components/amplify-totp-setup/amplify-totp-setup.js": "7 KB",
  "collection/components/amplify-username-field/amplify-username-field.js": "6 KB",
  "collection/components/amplify-verify-contact/amplify-verify-contact.js": "5 KB",
  "collection/global/theme.js": "2 KB",
  "collection/index.js": "111 Bytes",
  "docs.d.ts": "2 KB",
  "docs.json": "154 KB",
  "esm-es5/Translations-6c29ed51.js": "4 KB",
  "esm-es5/amplify-amazon-button_44.entry.js": "275 KB",
  "esm-es5/amplify-checkbox.entry.js": "1 KB",
  "esm-es5/amplify-federated-sign-in.entry.js": "2 KB",
  "esm-es5/amplify-greetings.entry.js": "2 KB",
  "esm-es5/amplify-nav_2.entry.js": "5 KB",
  "esm-es5/amplify-ui-components.mjs": "7 KB",
  "esm-es5/app-globals-66d47c1b.js": "2 KB",
  "esm-es5/auth-types-e7cd1f35.js": "2 KB",
  "esm-es5/constants-e5b85895.js": "2 KB",
  "esm-es5/css-shim-63363a96.js": "12 KB",
  "esm-es5/dom-8488c424.js": "27 KB",
  "esm-es5/helpers-4c8264ff.js": "1 KB",
  "esm-es5/index-f2c66794.js": "75 KB",
  "esm-es5/index.mjs": "123 Bytes",
  "esm-es5/loader.mjs": "7 KB",
  "esm-es5/shadow-css-779fc2c3.js": "16 KB",
  "esm/Translations-6c29ed51.js": "4 KB",
  "esm/amplify-amazon-button_44.entry.js": "227 KB",
  "esm/amplify-checkbox.entry.js": "1011 Bytes",
  "esm/amplify-federated-sign-in.entry.js": "2 KB",
  "esm/amplify-greetings.entry.js": "2 KB",
  "esm/amplify-nav_2.entry.js": "2 KB",
  "esm/amplify-ui-components.mjs": "6 KB",
  "esm/app-globals-66d47c1b.js": "2 KB",
  "esm/auth-types-e7cd1f35.js": "2 KB",
  "esm/constants-e5b85895.js": "2 KB",
  "esm/css-shim-63363a96.js": "10 KB",
  "esm/dom-8488c424.js": "19 KB",
  "esm/helpers-4c8264ff.js": "1 KB",
  "esm/index-f2c66794.js": "69 KB",
  "esm/index.mjs": "123 Bytes",
  "esm/loader.mjs": "7 KB",
  "esm/polyfills/core-js.js": "78 KB",
  "esm/polyfills/css-shim.js": "10 KB",
  "esm/polyfills/dom.js": "20 KB",
  "esm/polyfills/es5-html-element.js": "670 Bytes",
  "esm/polyfills/index.js": "1 KB",
  "esm/polyfills/promise.js": "7 KB",
  "esm/polyfills/system.js": "4 KB",
  "esm/shadow-css-779fc2c3.js": "15 KB",
  "index.js": "48 Bytes",
  "index.mjs": "36 Bytes",
  "loader/cdn.js": "124 Bytes",
  "loader/index.cjs.js": "124 Bytes",
  "loader/index.d.ts": "688 Bytes",
  "loader/index.es2017.mjs": "79 Bytes",
  "loader/index.mjs": "754 Bytes",
  "loader/node-main.js": "153 Bytes",
  "loader/package.json": "291 Bytes",
  "types/common/Translations.d.ts": "3 KB",
  "types/common/constants.d.ts": "1 KB",
  "types/common/country-dial-codes.d.ts": "110 Bytes",
  "types/common/helpers.d.ts": "631 Bytes",
  "types/common/testing.d.ts": "825 Bytes",
  "types/common/types/auth-types.d.ts": "3 KB",
  "types/common/types/ui-types.d.ts": "189 Bytes",
  "types/components.d.ts": "79 KB",
  "types/components/amplify-amazon-button/amplify-amazon-button.d.ts": "631 Bytes",
  "types/components/amplify-auth-fields/amplify-auth-fields-interface.d.ts": "534 Bytes",
  "types/components/amplify-auth-fields/amplify-auth-fields.d.ts": "916 Bytes",
  "types/components/amplify-auth-fields/component-field-mapping.d.ts": "393 Bytes",
  "types/components/amplify-auth0-button/amplify-auth0-button.d.ts": "400 Bytes",
  "types/components/amplify-authenticator/amplify-authenticator.d.ts": "949 Bytes",
  "types/components/amplify-button/amplify-button.d.ts": "427 Bytes",
  "types/components/amplify-checkbox/amplify-checkbox.d.ts": "455 Bytes",
  "types/components/amplify-code-field/amplify-code-field.d.ts": "1022 Bytes",
  "types/components/amplify-confirm-sign-in/amplify-confirm-sign-in.d.ts": "2 KB",
  "types/components/amplify-confirm-sign-up/amplify-confirm-sign-up.d.ts": "2 KB",
  "types/components/amplify-country-dial-code/amplify-country-dial-code-interface.d.ts": "166 Bytes",
  "types/components/amplify-country-dial-code/amplify-country-dial-code.d.ts": "473 Bytes",
  "types/components/amplify-email-field/amplify-email-field.d.ts": "833 Bytes",
  "types/components/amplify-examples/amplify-examples.d.ts": "116 Bytes",
  "types/components/amplify-examples/auth-totp-example.d.ts": "129 Bytes",
  "types/components/amplify-examples/authenticator-example.d.ts": "91 Bytes",
  "types/components/amplify-examples/custom-authenticator-example.d.ts": "97 Bytes",
  "types/components/amplify-examples/example4.d.ts": "97 Bytes",
  "types/components/amplify-examples/rock-paper-scissor/rock-paper-scissor.d.ts": "361 Bytes",
  "types/components/amplify-examples/scene-example.d.ts": "97 Bytes",
  "types/components/amplify-facebook-button/amplify-facebook-button.d.ts": "683 Bytes",
  "types/components/amplify-federated-buttons/amplify-federated-buttons.d.ts": "541 Bytes",
  "types/components/amplify-federated-sign-in/amplify-federated-sign-in.d.ts": "298 Bytes",
  "types/components/amplify-forgot-password/amplify-forgot-password-interface.d.ts": "56 Bytes",
  "types/components/amplify-forgot-password/amplify-forgot-password.d.ts": "2 KB",
  "types/components/amplify-form-field/amplify-form-field.d.ts": "2 KB",
  "types/components/amplify-form-section/amplify-form-section.d.ts": "720 Bytes",
  "types/components/amplify-google-button/amplify-google-button.d.ts": "750 Bytes",
  "types/components/amplify-greetings/amplify-greetings.d.ts": "626 Bytes",
  "types/components/amplify-hint/amplify-hint.d.ts": "56 Bytes",
  "types/components/amplify-icon-button/amplify-icon-button.d.ts": "198 Bytes",
  "types/components/amplify-icon/amplify-icon.d.ts": "233 Bytes",
  "types/components/amplify-icon/icons.d.ts": "335 Bytes",
  "types/components/amplify-input/amplify-input.d.ts": "1 KB",
  "types/components/amplify-label/amplify-label.d.ts": "78 Bytes",
  "types/components/amplify-link/amplify-link.d.ts": "95 Bytes",
  "types/components/amplify-loading-spinner/amplify-loading-spinner.d.ts": "66 Bytes",
  "types/components/amplify-nav/amplify-nav.d.ts": "55 Bytes",
  "types/components/amplify-oauth-button/amplify-oauth-button.d.ts": "211 Bytes",
  "types/components/amplify-password-field/amplify-password-field.d.ts": "1021 Bytes",
  "types/components/amplify-phone-field/amplify-phone-field.d.ts": "1023 Bytes",
  "types/components/amplify-radio-button/amplify-radio-button.d.ts": "772 Bytes",
  "types/components/amplify-require-new-password/amplify-require-new-password.d.ts": "1 KB",
  "types/components/amplify-scene-loading/amplify-scene-loading.d.ts": "235 Bytes",
  "types/components/amplify-scene/amplify-scene-interface.d.ts": "83 Bytes",
  "types/components/amplify-scene/amplify-scene.d.ts": "658 Bytes",
  "types/components/amplify-section/amplify-section.d.ts": "98 Bytes",
  "types/components/amplify-select-mfa-type/amplify-select-mfa-type.d.ts": "707 Bytes",
  "types/components/amplify-select/amplify-select-interface.d.ts": "416 Bytes",
  "types/components/amplify-select/amplify-select.d.ts": "594 Bytes",
  "types/components/amplify-sign-in-button/amplify-sign-in-button.d.ts": "132 Bytes",
  "types/components/amplify-sign-in/amplify-sign-in-interface.d.ts": "311 Bytes",
  "types/components/amplify-sign-in/amplify-sign-in.d.ts": "2 KB",
  "types/components/amplify-sign-out/amplify-sign-out.d.ts": "371 Bytes",
  "types/components/amplify-sign-up/amplify-sign-up-interface.d.ts": "337 Bytes",
  "types/components/amplify-sign-up/amplify-sign-up.d.ts": "2 KB",
  "types/components/amplify-strike/amplify-strike.d.ts": "58 Bytes",
  "types/components/amplify-toast/amplify-toast.d.ts": "181 Bytes",
  "types/components/amplify-tooltip/amplify-tooltip.d.ts": "268 Bytes",
  "types/components/amplify-totp-setup/amplify-totp-setup-interface.d.ts": "55 Bytes",
  "types/components/amplify-totp-setup/amplify-totp-setup.d.ts": "1 KB",
  "types/components/amplify-username-field/amplify-username-field.d.ts": "839 Bytes",
  "types/components/amplify-verify-contact/amplify-verify-contact.d.ts": "627 Bytes",
  "types/global/theme.d.ts": "125 Bytes",
  "types/index.d.ts": "109 Bytes",
  "types/stencil-public-runtime.d.ts": "53 KB",
}
`;
